modin/engines/ray/cudf_on_ray/frame/gpu_manager.py

```diff
@@ -62,21 +147,91 @@ def reduce(self, first, others, func, axis=0, **kwargs):
         return self.store_new_df(result)
 
     def store_new_df(self, df):
+        """
+        Store a new cudf_dataFrame in the dataframe_dict.
+        Iterate the current key int, and store the dataFrame in the dict
```

Comment:
I am not sure whether we are iterating per se. I would say something along the lines that we are saving a cudf.DataFrame in the next available unique key and returning the key